src/python/pants/backend/jvm/targets/jar_dependency.py

```diff
@@ -51,16 +111,34 @@ def __new__(cls, org, name, rev=None, force=False, ext=None, url=None, apidocs=N
       the dependency itself should be downloaded and placed on the classpath
     :param list excludes: Transitive dependencies of this jar to exclude.
     :type excludes: list of :class:`pants.backend.jvm.targets.exclude.Exclude`
+    :param string base_path: absolute base path that a relative file url is based from.
     """
     excludes = JarDependency._prepare_excludes(excludes)
+    base_path = base_path or get_buildroot()
+    if not os.path.isabs(base_path):
+      raise ValueError('base_path is not absolute: {}'.format(base_path))
     return super(JarDependency, cls).__new__(
         cls, org=org, base_name=name, rev=rev, force=force, ext=ext, url=url, apidocs=apidocs,
-        classifier=classifier, mutable=mutable, intransitive=intransitive, excludes=excludes)
+        classifier=classifier, mutable=mutable, intransitive=intransitive, excludes=excludes,
+        base_path=base_path)
 
   @property
   def name(self):
     return self.base_name
 
+  @memoized_method
+  def get_url(self, relative=False):
```

Comment:
Rather than renaming this, would recommend renaming the existing field to `url_`, and then making this a `@memoized_property def url`.

Or, could leave this method, and have the `url` property default to `relative=False`.